actionview/lib/action_view/template.rb

```diff
@@ -222,6 +227,24 @@ def encode!
       end
     end
 
+    # This method is responsible for marking a template as having explicit locals
+    # and extracting any arguments declared in the format
+    # locals: (message:, label: "My Message")
+    def explicit_locals!
+      source = self.source
+
+      # Check for match separately from replacement,
+      # as .sub!, while more memory efficient, does
+      # not work with frozen strings.
+      return unless source.match(EXPLICIT_LOCALS_REGEX)
```

Comment:
I opted for this approach so that existing templates that are frozen will continue to work as-is.

Reply:
In what case would the template be frozen?

Also if needed we can just dup it because it's an instance variable of template.

Reply:
Quite a few tests failed without this, likely because they set `frozen_string_literal: true` mocked in templates.

Reply:
Ah right, you can pass the source directly as argument, in which case it's likely a literal hence frozen.

I feel like we should just dup the string once in the constructor or something.